backend/Dockerfile

```diff
@@ -1,17 +1,19 @@
-FROM node:8.3.0-alpine
+FROM node:alpine
 MAINTAINER mastersprintr@gmail.com
 
-# Obnoxious fix for a very weird error
-# RUN apk add --update nodejs nodejs-npm && npm install npm@latest -g
-
-ENV http_proxy=http://10.138.15.10:8080 https_proxy=http://10.138.15.10:8080
+# ENV http_proxy=http://10.138.15.10:8080 https_proxy=http://10.138.15.10:8080
 
 RUN apk update && apk add bash
 
-RUN npm config set proxy http://10.138.15.10:8080
-RUN npm config set https_proxy http://10.138.15.10:8080
+# RUN npm config set proxy http://10.138.15.10:8080
+# RUN npm config set https_proxy http://10.138.15.10:8080
+
+RUN mkdir -p /app/dist
+ADD package.json /tmp/package.json
+RUN cd /tmp && npm install
+RUN cp -a tmp/node_modules/ /app/node_modules/
```

Comment:
is there a reason why we install node_modules on a separate directory? can't we just install on /app?